sdks/java/io/solace/src/main/java/org/apache/beam/sdk/io/solace/write/UnboundedSolaceWriter.java

```diff
@@ -165,16 +173,27 @@ public void publishResults(BeamContextWrapper context) {
           maxFailed = Math.max(maxFailed, latency);
         }
       }
-
       if (result.getPublished()) {
         context.output(
-            SUCCESSFUL_PUBLISH_TAG, result, getCurrentBundleTimestamp(), getCurrentBundleWindow());
+            SUCCESSFUL_PUBLISH_TAG, result, getCurrentBundleTimestamp(), GlobalWindow.INSTANCE);
       } else {
-        context.output(
-            FAILED_PUBLISH_TAG, result, getCurrentBundleTimestamp(), getCurrentBundleWindow());
+        try {
+          BadRecord b =
+              BadRecord.fromExceptionInformation(
+                  result,
+                  null,
+                  null,
+                  result.getError() != null
+                      ? checkNotNull(result.getError())
+                      : "SolaceIO.Write: unknown error.");
```

Comment:
Create a new variable, assign it `result.getError()` and then perform this null check against the variable instead. The reason why error prone keeps complaining about it is because the two calls two `getError` may return two different results, storing the return value is what you want here. The other thing you could do is `Optional.ofNullable(result.getError()).orElse("SolaceIO.Write: unknown error.")`.

Reply:
Good point, addressed.